llvm/lib/Transforms/InstCombine/InstCombineCompares.cpp

```diff
@@ -8037,6 +8037,45 @@ Instruction *InstCombinerImpl::visitFCmpInst(FCmpInst &I) {
       if (Instruction *NV = FoldOpIntoSelect(I, cast<SelectInst>(LHSI)))
         return NV;
       break;
+    case Instruction::FSub:
+      switch (Pred) {
+      default:
+        break;
+      case FCmpInst::FCMP_UGT:
+      case FCmpInst::FCMP_ULT:
+      case FCmpInst::FCMP_UNE:
+      case FCmpInst::FCMP_OEQ:
+      case FCmpInst::FCMP_OGE:
+      case FCmpInst::FCMP_OLE:
+        // Skip optimization: fsub x, y unless guaranteed !isinf(x) && 
+        // !isinf(y).
```

Comment:
```suggestion
        // Skip optimization: fsub x, y unless guaranteed !isinf(x) || 
        // !isinf(y).
```
The only problematic case is `inf - inf = nan`.